<a href="https://colab.research.google.com/github/werowe/HypatiaAcademy/blob/master/numpy/November_7_scikit_learn_linear_regression.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import matplotlib.pyplot as plt
from sklearn import linear_model
import numpy as np
from sklearn.metrics import mean_squared_error, r2_score

reg = linear_model.LinearRegression(fit_intercept=True)

data = np.array([[1, 2, 3, 4, 5, 6, 7],[1.8, 4.3, 6.2,  5, 11, 11, 15]])

x = data[0,:].reshape(7,1)
y = data[1,:]


reg.fit(x,y)

print('Coefficients: \n', reg.coef_)
print('Intercept: \n', reg.intercept_)
print('Score: \n' , reg.score(x, y))
print('Mean squared error: %.2f' % mean_squared_error(x, y))
print('Coefficient of determination: %.2f' % r2_score(x, y))

Coefficients: 
 [2.06428571]
Intercept: 
 -0.4999999999999991
Score: 
 0.9111050507254281
Mean squared error: 20.31
Coefficient of determination: -4.08


So, your function is y = 2.06428571x - 0.4999999999999991
Remember that ML uses the normal curve in expressing errors.

Mean Squared Error (MSE) is used here.

The **coefficient of determination** (R2) is the proportion of variance that can be explained by the independent variables. If it is 0, then the two are not in any way correlated.

In [20]:
print('Mean squared error: %.2f' % mean_squared_error(y, reg.predict(x)))


Mean squared error: 1.66


Homework.  Please figure out why in the spreadsheet I have the wrong answer for MSE.

https://docs.google.com/spreadsheets/d/1YHy2Th3Vmc0Ye3bt21YnTtoygWtl927TkyS1wMRTep0/edit#gid=0